### PR TITLE
Security & Extra Linting

### LINKED ISSUE #340: Error: GitHub Repository Not Mapped To eMASS System
The Department of Veterans Affairs (VA) Office of Information Security (OIS) has detected that the following repository has not been mapped to an eMASS system:

Code scanning is not enabled for the following source code repository:
- https://github.com/department-of-veterans-affairs/healthcare-application

VA policy requires mapping all repositories in the GitHub Enterprise Cloud to the eMASS system that the application belongs to.

Our automation has analyzed your repository and identified the following [supported CodeQL languages](https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/#languages-and-compilers):
- `javascript`

The steps that you will need to take can be found in the following technical note:
- [How to update the emass.json file](https://department-of-veterans-affairs.github.io/ois-swa-wiki/docs/ghas/codeql-usage/update-emass-json/)

To request technical support: 
- [Please book a meeting with OIS ISO SAVD Software Assurance](https://outlook.office365.com/owa/calendar/OISSoftwareAssurance@DVAGOV.onmicrosoft.com/bookings/)

VA OIT Logo

### PR DESCRIPTION
**Pull request by markolson**
- Adds [NSP](https://github.com/nodesecurity/nsp), which scans dependencies to see if they are versions that have known vulnerabilities
- Adds many more ESLint rules from the 18F recommendation to use scanjs-rules

Closes #340

### CODE REVIEW

**Comment by akainic** [comment]
@robbiethegeek do you mind reviewing this one?

File: package.json [hunk 1]
```diff
@@ -93,6 +96,7 @@
     "moment": "^2.13.0",
     "morgan": "^1.7.0",
     "node-sass": "^3.6.0",
+    "nsp": "^2.4.0",
```
Comment (robbiethegeek) [inline package.json]: Are we running this on Travis and deployment?  I am worried that we are installing a CLI that likely needs to be a local binary that will slow down things.

**Comment by robbiethegeek** [comment]
Seems like our friends at Sauce Labs are tripping this up again, not sure what the error is it led me to https://wiki.saucelabs.com/display/DOCS/Common+Error+Messages
Retrying and hoping it passes if not we can look further.

**Comment by alexose** [comment]
Despite the "Uh oh, you've run out of minutes" error, I'm guessing the Saucelabs things was a bug on their end.  Open Source accounts aren't supposed to have minutes for automated tests...

I pinged them to see what's up.

File: package.json (hunk 1 above)
Comment (robbiethegeek) [inline package.json]: here is the tldr of the above comment :) @markolson maybe move this to dev dependencies since I don't think we need this on deploy

**Pull request merged**